Assets/Scripts/UI/GalleryMenu.cs

```diff
@@ -0,0 +1,221 @@
+                foreach (var renderer in gun.GetComponentsInChildren<Renderer>())
+                {
+                    renderer.gameObject.layer = LayerMask.NameToLayer("UI");
+                    bounds.Encapsulate(renderer.bounds);
+                }
+
+                // Scale the weapon so it fits within the UI 
+                float scaleFactor = gridCellSize.x / bounds.size.z;
+
+                gun.transform.localScale = new Vector3(scaleFactor / gun.transform.lossyScale.x, scaleFactor / gun.transform.lossyScale.y, scaleFactor / gun.transform.lossyScale.z);
+
+                // Recalculate bounds
+                foreach (var renderer in gun.GetComponentsInChildren<Renderer>())
+                {
+                    renderer.gameObject.layer = LayerMask.NameToLayer("UI");
+                    bounds.Encapsulate(renderer.bounds);
+                }
+
+                // Center the gun on the grid cell
+                gun.transform.localPosition = new Vector3(0 - bounds.center.z * gridCellSize.x / 4, gridCellSize.y / 3, -bounds.extents.x);
+
+                // Rotate the weapon to the correct angle
+                gun.transform.Rotate(Vector3.up * 90);
+
+                // Add the weapon name
+                string name = GunFactory.GetGunName(weapon.body, weapon.barrel, weapon.extension);
+                gun.name = name;
+                gridElement.GetComponentInChildren<TMP_Text>().text = name;
+            } else
+            {
+                // We are out of displayable weapons, disable objects
+                gridElement.gameObject.SetActive(false);
+            }    
+        }
+    }
+
+    public void NextPage(InputAction.CallbackContext ctx)
+    {
+        if(pageIndex == maxPages - 1)
+        {
+            pageIndex = 0;
+        }
+        else
+        {
+            pageIndex++;
+        }
+
+        selectedTab.GetComponent<Image>().color = disabledColor;
+        selectedTab = tabs[pageIndex];
+        selectedTab.GetComponent<Image>().color = selectedColor;
+
+        PopulateGrid(pageIndex);
+    }
+
+    public void PrevPage(InputAction.CallbackContext ctx)
+    {
+        if (pageIndex == 0)
+        {
+            pageIndex = maxPages - 1;
+        }
+        else
+        {
+            pageIndex--;
+        }
+
+        selectedTab.GetComponent<Image>().color = disabledColor;
+        selectedTab = tabs[pageIndex];
+        selectedTab.GetComponent<Image>().color = selectedColor;
+
+        PopulateGrid(pageIndex);
+    }
+
+    public void Back(InputAction.CallbackContext ctx)
+    {
+        mainMenuController.SwitchToMenu(mainMenuController.defaultMenu);
+        mainMenuController.SelectControl(mainMenuController.GetComponentInChildren<Button>());
+    }
+
+    /// <summary>
+    /// Creates and returns an array of all possible weapon combinations
+    /// </summary>
+    private Weapon[] CreateAllWeapons()
+    {
+        StaticInfo sInfo = StaticInfo.Singleton;
```

Comment:
This collection of structs does not impact performance in any meaningful way and is also only a placeholder function before we implement ways of storing gun statistics and part performance as well as legendary weapons. So I feel this can stay.

Reply:
Um, I was talking about the `sInfo` abbreviation

Reply:
You got to be more specific man! But I've renamed the variable name properly now.

Reply:
Sorry, I thought it'd be evident from the line I marked 🙃